c-sharp-server/c-sharp-client/Program.cs

```diff
@@ -60,41 +79,57 @@ class App {
                     case String s when command.StartsWith("--login"):
                         // the password is jellybeans
                         string password = s.Replace("--login ", "");
-                        var res = ditto.Auth.LoginWithToken(password, "provider");
+                        await ditto.Auth.LoginWithToken(password, "provider");
                         break;
+
+                    case String s when command.StartsWith("--logout"):
+                        ditto.Auth.Logout();
+                        break;
+
                     case string s when command.StartsWith("--insert"):
                         string taskBody = s.Replace("--insert ", "");
-                        ditto.Store["tasks"].Upsert(new Task(taskBody, false).ToDictionary());
+                        var task = new Task(taskBody, false).ToDictionary();
+                        await ditto.Store.ExecuteAsync("INSERT INTO tasks DOCUMENTS (:task)", new Dictionary<string, object> { { "task", task } });
                         break;
+
                     case string s when command.StartsWith("--toggle"):
                         string _idToToggle = s.Replace("--toggle ", "");
-                        ditto.Store["tasks"]
-                            .FindById(new DittoDocumentId(_idToToggle))
-                            .Update((mutableDoc) => {
-                                if (mutableDoc == null) return;
-                                mutableDoc["isCompleted"].Set(!mutableDoc["isCompleted"].BooleanValue);
-                            });
+                        var isCompleted = tasks.First(t => t._id == _idToToggle).isCompleted;
+                        await ditto.Store.ExecuteAsync(
+                            "UPDATE tasks " +
+                            "SET isCompleted = :newValue " +
+                            "WHERE _id == :id",
+                            new Dictionary<string, object> {{ "newValue", !isCompleted }, { "id", _idToToggle }});
```

Comment:
nit: I would use string interpolation since it'd be easier to read. 

```suggestion
                        var isCompleted = tasks.First(t => t._id == _idToToggle).isCompleted;
                        await ditto.Store.ExecuteAsync(
                            "UPDATE tasks " +
                            $"SET isCompleted = {!isCompleted} " +
                            $"WHERE _id == {_idToToggle}"); 
```

Reply:
Similar to [this comment](https://github.com/getditto/template-app-swift-todo/pull/1#discussion_r1418274405), but it would be nice to show how to use query arguments to developers.